### PR TITLE
PR: Bump pylint to `>=3.1,4` and python-lsp-server to `>=1.11.0,<1.12.0`

### LINKED ISSUE #21936: Release 5.5.4
This is an issue to track changes regarding dependencies done for the new Spyder 5.5.4 release.

* New/updated dependencies:
     * `qtawesome` >=1.3.1,<1.4.0 (see #21939)
     * `python-lsp-server` >=1.11.0,<1.12.0 + `pylint` >=3.1,<4 (see #21942)

### PR DESCRIPTION
**Pull request by bnavigator**
## Description of Changes

Bump Pylint and python-lsp-server to the new versions

### Issue(s) Resolved

Fixes #21936 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: bnavigator

### CODE REVIEW

**Comment by bnavigator** [comment]
Bumped pylsp because of python-lsp/python-lsp-server#543

File: setup.py [hunk 1]
```diff
@@ -226,13 +226,13 @@ def run(self):
...
     'pygments>=2.0',
-    'pylint>=2.5.0,<3.1',
+    'pylint>=3.1,<4',
     'pylint-venv>=3.0.2',
     'python-lsp-black>=2.0.0,<3.0.0',
     'pyls-spyder>=0.4.0',
     'pyqt5>=5.10,<5.16',
     'pyqtwebengine>=5.10,<5.16',
-    'python-lsp-server[all]>=1.10.0,<1.11.0',
+    'python-lsp-server[all]>=1.11.0,<1.12.0',
```
Comment (ccordoba12) [inline setup.py:235]: Please update the constraint for development below to be

```python
install_requires.append('python-lsp-server[all]>=1.11.0,<1.13.0')
```

**Review by ccordoba12 — DISMISSED** [review]
@bnavigator, besides addressing my comment below, you also need to sync our subrepo for `python-lsp-server` (i.e. a copy we have of that project here).

For that, please follow [these instructions](https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md#making-contributions-that-depend-on-pull-requests-in-python-lsp-server-or-qtconsole), specifically this command:

    git subrepo pull external-deps/python-lsp-server -r https://github.com/python-lsp/python-lsp-server.git -b develop -u -f

File: setup.py (hunk 1 above)
Comment (bnavigator) [reply setup.py]: Done

Comment (bnavigator) [reply setup.py:235]: Now really. Sorry I didn't read your instruction thoroughly.

**Review by ccordoba12 — APPROVED** [review]
Looks good to me now, thanks @bnavigator!

Note: The failure in our tests was fixed in PR #21939.

**Pull request merged**